ci/docker/ubuntu-22.04-cpp.dockerfile

```diff
@@ -115,10 +116,20 @@ RUN apt-get update -y -q && \
         rapidjson-dev \
         rsync \
         tzdata \
-        wget && \
+        wget \
+        xz-utils && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists*
 
+# install emscripten using EMSDK
+ARG emscripten_version="3.1.45"
+RUN cd ~ && git clone https://github.com/emscripten-core/emsdk.git && \
```

Comment:
Hmm, two things:
1. it would have been better for resource usage to make a separate dockerfile for this, perhaps
2. could this be a script to avoid having tons of hand-coded bash code in dockerfiles?